File: src/main/java/com/dxschool/lightme/caseuser/controller/dto/CaseUserResponse.java
package com.dxschool.lightme.caseuser.controller.dto;

import com.dxschool.lightme.caseuser.domain.CaseUser;

public record CaseUserResponse(
    String artist,
    int brightness,
    String colorCode,
    long point,
    String profileUrl,
    String nickname,
    AddressResponse address,
    String intro
) {
    public static CaseUserResponse from(CaseUser caseUser) {
        return new CaseUserResponse(
                caseUser.getThemeArtist().getArtistName(),
                caseUser.getBrightness(),
                caseUser.getColor(),
                caseUser.getPoint(),
                caseUser.getProfileLink(),
                caseUser.getNickname(),
                AddressResponse.from(caseUser.getAddress()),
                caseUser.getIntro()
        );
    }
}

File: android/app/src/main/java/com/example/tfisland/model/dto/CaseUserResponse.kt
package com.example.tfisland.model.dto

data class CaseUserResponse(
    val artist: String,
    val address: AddressResponse,
    val brightness: Int,
    val colorCode: String,
    val point: Long,
    val intro: String,
    val profileLink: String?,
    val nickname: String
)
